From 9200e45d21f2f2f88fc5fd65f988dfc8d072d48a Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@freescale.com>
Date: Tue, 27 Oct 2015 02:30:47 +0800
Subject: [PATCH] MLK-11762 ARM: imx: disable SNVS clk to save power when
 resume

SNVS clk now is enabled/disabled dynamically in order to save
SNVS domain power, before suspend, its clk is enabled for
register access, better to disable it after resume to make
sure SNVS power is low enough in case there is no SNVS clk
disable during driver resume phase.

Signed-off-by: Anson Huang <Anson.Huang@freescale.com>
---
 arch/arm/mach-imx/suspend-imx7.S | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index 34910ad..ea07123 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -551,6 +551,12 @@ wfi_ddr_retention_out:
 
 	restore_ttbr1
 standby_out:
+	/* make sure SNVS clk is disabled */
+	ldr	r11, [r0, #PM_INFO_MX7_CCM_V_OFFSET]
+	add	r11, r11, #0x4000
+	ldr	r7, =0x0
+	str	r7, [r11, #CCM_SNVS_LPCG]
+
 	pop	{r4-r12}
 	/* return to suspend finish */
 	mov	pc, lr
@@ -585,6 +591,11 @@ resume:
 dsm_ddr_self_refresh_out:
 	ddrc_exit_self_refresh
 dsm_ddr_retention_out:
+	/* make sure SNVS clk is disabled */
+	ldr	r11, [r0, #PM_INFO_MX7_CCM_P_OFFSET]
+	add	r11, r11, #0x4000
+	ldr	r7, =0x0
+	str	r7, [r11, #CCM_SNVS_LPCG]
 
 	mov	pc, lr
 ENDPROC(imx7_suspend)
-- 
1.8.0

